routing/result_interpretation.go

```diff
@@ -367,7 +367,19 @@ func (i *interpretedResult) processPaymentOutcomeUnknown(route *route.Route) {
 // failNode marks the node indicated by idx in the route as failed. This
 // function intentionally panics when the self node is failed.
 func (i *interpretedResult) failNode(rt *route.Route, idx int) {
+	// Mark the node as failing.
 	i.nodeFailure = &rt.Hops[idx-1].PubKeyBytes
+
+	// Mark the incoming connection as failed for the node.
+	inPair, _ := getPair(rt, idx-1)
+	i.pairResults[inPair.Reverse()] = pairResult{}
```

Comment:
If I remember correctly, there were comments in other prs about not adding redundant initializers. Not sure what you standard is here.

Reply:
Personally I like adding them (exceptions being `nil` initializers perhaps), and in this case I feel it makes sense, since an empty struct is not obvious what it means.